src/app/directives/to-positive-number.directive.ts

```diff
@@ -5,24 +5,13 @@ import { Directive, HostListener } from '@angular/core';
 })
 export class ToPositiveNumberDirective {
 
-  private lastValue: any;
-
-  @HostListener( 'keyup', [ '$event' ] )
-  numberPositive( ev: any ) {
-    if (!this.isValidKey(ev.key)) {
-      ev.currentTarget.value = this.lastValue;
+  @HostListener( 'keydown' || 'paste', [ '$event' ] )
```

Comment:
I had separated each event handler in a different method in order to make it easier to extend them in the future.